gobblin-runtime/src/main/java/org/apache/gobblin/runtime/job_catalog/NonObservingFSJobCatalog.java

```diff
@@ -121,5 +125,11 @@ public synchronized void remove(URI jobURI) {
       LOGGER.warn("No file with URI:" + jobURI + " is found. Deletion failed.");
     }
   }
+
+  @Override
+  public synchronized void cancel(JobSpec jobSpec) {
+    Preconditions.checkState(state() == State.RUNNING, String.format("%s is not running.", this.getClass().getName()));
```

Comment:
I am confused by this: Check the state equals to running but log "xxx is not running" ?

Reply:
Yea, the message is supposed to be "exception message to use if the check fails" (from Preconditions# checkState () javadoc)